test/sanitytest.c

```diff
@@ -76,12 +76,20 @@ static int test_sanity_unsigned_convertion(void)
     return 1;
 }
 
+static int test_sanity_pointer_sizes(void)
+{
+    if (!TEST_int_eq(sizeof(long), sizeof(void *)))
+        return 0;
+    return 1;
+}
+
 void register_tests(void)
 {
     ADD_TEST(test_sanity_null_zero);
     ADD_TEST(test_sanity_enum_size);
     ADD_TEST(test_sanity_twos_complement);
     ADD_TEST(test_sanity_sign);
     ADD_TEST(test_sanity_unsigned_convertion);
+    ADD_TEST(test_sanity_pointer_sizes);
```

Comment:
I suppose question also is how would LONG work now [on sizeof(long)!=sizeof(void*) platform in case application chooses to], right? I mean even if it's deprecated, it doesn't mean that it's appropriate to disregard it till it's actually gone. But either way, suggested check is unjustified.